index.d.ts

```diff
@@ -230,6 +230,23 @@ declare module '@primer/components' {
   export const Sticky: React.FunctionComponent<PositionComponentProps>
   export const Fixed: React.FunctionComponent<PositionComponentProps>
 
+  export interface SideNavProps extends CommonProps, BorderProps, Omit<React.HTMLAttributes<HTMLDivElement>, 'color'> {
```

Comment:
Adding `BorderProps` here means that someone who is using TS would see that the border system props are available even when they aren't passing in the `bordered` boolean prop. Another reason why we should consider always giving access to the `BorderBox` props and just using `bordered` to set default styles. I think we'd need to use `Box` and pass in the border system props separately instead of using a `BorderBox`.